### DISCUSSION TITLE
Insert code is going wrong

### DISCUSSION THREAD

**Discussion by Schrijvers123** [joomengine/Joomla-Component-Builder#924] [Extention Issues]
I add this to the controller file of my compnent:
```
// No direct access to this file
defined('_JEXEC') or die('Restricted access');

/***[INSERTED$$$$]***//**144**/
require_once (JPATH_COMPONENT_ADMINISTRATOR.'/helpers/hostfact_api.php');
/***[/INSERTED$$$$]***/
```

after compiling it compiled to:
```
/***[INSERTED$$$$]***//**144**/
// require_once (JPATH_COMPONENT_ADMINISTRATOR.'/helpers/hostfact_api.php');
// 
/***[/INSERTED$$$$]***/
```
What did I wrong? If i upload the code bu ftp it;s working again. What am I doing wrong?

**Comment by Llewellynvdm** [comment]
There are many things that can effect this, I would suggest not using custom code for this, but the header plugins that target the header areas of JCB files.

I am going to push all of JCB PRO public in a few days and then you will be able to do so much more... until then the current public version has the plugins as part of its package... if you install this package you will get access to all the header areas that are already public.

Using the custom code option must always be the last resort as any change to the code above the inserted code "if it changes" will cause JCB to lose track of where to place it, and so 8-10 lines above this code and below this code if it changes between compilations, JCB can not know where to place the code and so will escape it to prevent it being placed in the wrong place.

**Reply by Schrijvers123** [reply]
I thought i have the pro version (as donator). Where can I find that then?

**Reply by Llewellynvdm** [reply]
If you are using the Pro Version of JCB you would also have seen all the videos that show how each of these features work, if not please watch them.

https://www.youtube.com/playlist?list=PLQRGFI8XZ_wsV6PDFXd21B7O0X70JlNd3

**Reply by Schrijvers123** [reply]
I did, and now i try to uses the header plugin, but son;t now where and how to put the reference 'helper/hostfact_api.php'

The reference needs to be from the controller places in administrator\componenten\[componentname]\controllers\trainingen.php.

As you can see in the image below ad added it.
![image](https://user-images.githubusercontent.com/3889668/177501921-7358d7da-abff-434c-9d04-e79a2b15931e.png)

What did I wrong? Please help (I saw the instruction movie: https://www.youtube.com/watch?v=TjoH5rj4UgM).
